tests/ui_tools/test_boxes.py

```diff
@@ -105,23 +105,25 @@ def test_autocomplete_mentions(self, write_box, users_fixture,
         assert typeahead_string == required_typeahead
 
     @pytest.mark.parametrize('text, state, required_typeahead', [
-        ('#Stream', 0, '#**Stream 1**'),
-        ('#Stream', 1, '#**Stream 2**'),
-        ('#S', 0, '#**Some general stream**'),
-        ('#S', 1, '#**Secret stream**'),
+        ('#Stream', 0, '#**Secret stream**'),
+        ('#Stream', 1, '#**Some general stream**'),
+        ('#Stream', 2, '#**Stream 1**'),
+        ('#Stream', 3, '#**Stream 2**'),
```

Comment:
Shouldn't this give the order of "Stream 1", "Stream 2", "Secret stream", "Some general stream" ?

Are there pinning considerations? If so, can we make that explicit in some way?

Reply:
@neiljp Yes, there are pinning considerations.
The 'pinned_streams' are defined, in `test_autocomplete_streams`, through the following:
```
write_box.view.pinned_streams = [
        [stream['name']] for stream in
        streams_fixture[:len(streams_fixture)//2]
```
Would you like the function to be refactored?

Reply:
My concern is that these parametrizations don't demonstrate what happens very clearly.

I would expect test cases showing each of the features we claim to support, if that's possible with the stream names we have. So `#stream` would (without pinning) prefer stream-1, stream-2, secret, then general; `#S` would just be alphabetical (?); more matches limit things (like `#so`, `#se` do already :+1:).

A comment may be sufficient here, though we may need a pinning of specific named streams in the test to make it clearer too?